### ISSUE TITLE
apps.humanatlas.io/us1: Cell Population Predictor

### ISSUE THREAD

**Issue by LibbyUX** [hubmapconsortium/hra-ui#713]
# Ready for dev: Cosmetic updates for apps.humanatlas.io/us1: Cell Population Predictor
This app is responsive and received the screen size notice when widths are below 1280px. Min/max widths are placed around this flow intentionally to provide the best user experience for varying breakpoints.
- [Canvas](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=1080-5348&t=NfOSydhkkdFKP10t-4)
- [Prototype](https://www.figma.com/proto/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=906-3752&node-type=canvas&viewport=997%2C4375%2C0.15&t=NfOSydhkkdFKP10t-0&scaling=scale-down&content-scaling=fixed&starting-point-node-id=1087%3A8498&show-proto-sidebar=1)

Please let me know if you have questions, thanks!

![image](https://github.com/user-attachments/assets/981965ac-1c9a-4cbb-a0c7-6e0077d1dcb9)

**Comment by LibbyUX** [comment]
Hi @andreasbueckle,

I received a question just now from @axdanbol:
> Another thing, if you check the preview here: https://67940fb6312428529f6e2207--hra-ui.netlify.app/apps/apps.humanatlas.io/result
The % of Total needs formatting. How should we format this? Especially since the numbers of zeros after the decimal point is different for each row. 

Daniel and @prashantchanne12 are currently building this UI. Could you please provide guidance on the formatting expectations for this percentage? Currently in the approved Figma specification [here](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=1515-15722&t=YR3G00boAUDh7eRr-4), I specified 5 decimal places.

# Preview build 1.27.2025
![Image](https://github.com/user-attachments/assets/8d73c390-b9bd-4f1f-93cc-d5df68c09026)

# Figma
- https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=1515-15722&t=YR3G00boAUDh7eRr-4

![Image](https://github.com/user-attachments/assets/f563cfb6-6c12-4118-aadc-6e4ab35fe0ed)

**Comment by andreasbueckle** [comment]
I checked out https://67940fb6312428529f6e2207--hra-ui.netlify.app/apps/apps.humanatlas.io/result. 

Are we sure we are already showing percentages? For example, as the highest number, we show: 
`0.13284199903812066%`. Don't you mean `13.284199903812066%`?

As for the very long decimal numbers, I suggestet scientific notation: https://www.calculatorsoup.com/calculators/math/scientific-notation-converter.php

`0.00000433948095483701` = `4.34 × 10-6`

**Comment by axdanbol** [comment]
> I checked out https://67940fb6312428529f6e2207--hra-ui.netlify.app/apps/apps.humanatlas.io/result.
> 
> Are we sure we are already showing percentages? For example, as the highest number, we show: `0.13284199903812066%`. Don't you mean `13.284199903812066%`?

Yes, I checked and we need to multiply it by 100

> 
> As for the very long decimal numbers, I suggestet scientific notation: https://www.calculatorsoup.com/calculators/math/scientific-notation-converter.php
> 
> `0.00000433948095483701` = `4.34 × 10-6`

Neither Bruce or I are big fans of scientific notation.
Another suggestions could be that for values such as the above one we show `< 0.01%` or `less than 0.01%`

**Comment by andreasbueckle** [comment]
I would try and keep these values in a format that can easily be converted to another one downsteam. Most people will download the CSV file and they might still want to do computation over the percentage column even if the numbers are tiny. This makes scientific notation better IMO as it is easier to convert to decimals and because it does not lose information.

**Comment by bherr2** [comment]
Yes, download should absolutely be the raw (unnormalized values) that come out of the API currently. However, display should be more easily read.

**Comment by LibbyUX** [comment]
Please let me know if anything needs to be updated in the design in Figma and I can make it happen!

**Comment by andreasbueckle** [comment]
> Yes, download should absolutely be the raw (unnormalized values) that come out of the API currently. However, display should be more easily read.

In this case, I suggest:
- For display, let's use `< 0.01%` (side note: will the user be able to sort by percentage? If so, we need to make sure this still shows up at the bottom when sorted in `DESC` order)
- For download, let's use raw numbers.

**Comment by LibbyUX** [comment]
Thank you, @andreasbueckle! We will do this.

@andreasbueckle, I have a couple more questions from ideating with @axdanbol just now:
- Which column header default sorts this table? Ascending or descending?
- For the columns `% of total` and `# cells`:
  - For integers, do we show trailing zeroes or not?

Just want to ensure our tables are formatted nicely for the UX and to both you and Katy's preference. Thanks for everything!

I mocked up some variations in Figma so you can see what the options look like for integers:

![Image](https://github.com/user-attachments/assets/d90bb8ec-ebc4-4eb2-8807-ce5a08d73ffd)

@axdanbol, please chime in if I missed anything!

**Comment by andreasbueckle** [comment]
% of total should sort, DESC by default. For integers, no trailing zeroes.

----------------------------------------------
Best,
Andreas

Andreas "Andi" Bueckle, Ph.D.
Research Lead, Cyberinfrastructure for Network Science Center (CNS)
Faculty Advisor, Sailing Club at Indiana University
US Sailing Certified Race Official
Room #4024
Luddy School of Informatics, Computing, and Engineering, Indiana University
700 N Woodlawn Ave (Luddy Hall)
47408 Bloomington, IN - USA

email: ***@***.******@***.***>
web: www.andreas-bueckle.com<https://www.andreas-bueckle.com/>
________________________________

**Comment by andreasbueckle** [comment]
OK, yes, better id width does not change between them. So, yes to trailing zeroes.

**Comment by LibbyUX** [comment]
Hi @prashantchanne12 & @axdanbol, thank you so much for discussing this app with me today! I left comments spatially in the canvas so you can see what has been tweaked!

Compiling a resource list for us all to stay organized here:

### Sections
- [Prior iteration](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=2367-22140&t=UXwtF6CcFA1YbWSs-4), before responsive edits
- [Current iteration](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=2343-13059&t=UXwtF6CcFA1YbWSs-4), after responsive edits
- - [Breakpoint example section](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=2367-24099&t=UXwtF6CcFA1YbWSs-4)

### Local components
- [Product Header](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=2343-20596&t=UXwtF6CcFA1YbWSs-4)
  - Side note: @axdanbol, let me know if I should request this as a global component. We will use it for the other apps in this file, too, but we currently are not using this pattern elsewhere. 
- [Cell Pop Predictor cards](https://www.figma.com/design/sASmFJEO5Usd3YS0IXNMBQ/apps.humanatlas.io?node-id=2343-20596&t=UXwtF6CcFA1YbWSs-4)

![Image](https://github.com/user-attachments/assets/8ade975a-b1af-4d3a-8f5f-108ff461402f)
